lib/srv/db/server.go

```diff
@@ -462,11 +465,25 @@ func (s *Server) getProxiedDatabases() (databases types.Databases) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 	for _, database := range s.proxiedDatabases {
-		databases = append(databases, database)
+		copy := database.Copy()
+		s.injectEC2Labels(copy)
+		databases = append(databases, copy)
 	}
 	return databases
 }
 
+func (s *Server) injectEC2Labels(database types.Database) {
```

Comment:
This logic repeats multiple times throughout your PR. Is there a way to encapsulate it somehow in the EC2Labels interface? Maybe have a `Apply()` methods on it or something that accepts `ResourceWithLabels` and sets auto-fetched labels to it?

If it complicates things much not a big deal, just curious if we can minimize code duplication a bit this way.

Reply:
@atburke Sorry just saw this. Instead of creating a new interface, would it make sense to extend ResourceWithLabels interface with Get/SetStaticLabels methods? Seems fitting.